### PR TITLE
lib/options: nullable mkPackageOption

### PR DESCRIPTION
**Pull request by ncfavier**
It is sometimes useful to allow setting a package option to `null` to skip installing the package. See https://github.com/nix-community/home-manager/pull/3668#issuecomment-1554044171 for example.

No manualHTML diff, haven't tested beyond that.

### CODE REVIEW

**Review by kubukoz — APPROVED** [review]

File: lib/options.nix [hunk 1]
```diff
@@ -164,19 +166,24 @@ rec {
...
-          + (if extraDescription == "" then "" else " ") + extraDescription;
-        ${if default != null then "default" else null} = defaultValue;
-        ${if example != null then "example" else null} = literalExpression
+      } else if nullable then {
+        default = null;
+      } else { }) // lib.optionalAttrs (example != null) {
+        example = literalExpression
           (if isList example then "pkgs." + concatStringsSep "." example else example);
-      };
+      });
```
Comment (roberth) [inline lib/options.nix:187]: This getting way too complicated to go without tests.

Comment (roberth) [reply lib/options.nix:187]: Please add tests in `lib/tests/modules.sh`

Comment (ncfavier) [reply lib/options.nix:187]: Here you go

**Review by roberth — APPROVED** [review]
Thanks!

**Pull request merged**